pkg/server/endpoints/registration/handler_test.go

```diff
@@ -744,6 +745,125 @@ func (s *HandlerSuite) TestListBySpiffeID() {
+	}{
+		{
+			name:         "without token and page size",
+			request:      &registration.ListAllEntriesRequest{},
+			expectedList: []*common.RegistrationEntry{entry3, entry2, entry1},
+		},
+		{
+			name: "with just page size",
+			request: &registration.ListAllEntriesRequest{
+				Pagination: &registration.Pagination{
+					PageSize: 2,
+				},
+			},
+			expectedList: []*common.RegistrationEntry{entry2, entry1},
+		},
+		{
+			name: "with request page size as 0 but returning list of default page size",
+			request: &registration.ListAllEntriesRequest{
+				Pagination: &registration.Pagination{
+					PageSize: 0,
+				},
+			},
+			expectedList:     []*common.RegistrationEntry{entry3, entry2, entry1},
+			expectedPageSize: defaultListEntriesPageSize,
+		},
+		{
+			name: "with request page size greater than entries",
+			request: &registration.ListAllEntriesRequest{
+				Pagination: &registration.Pagination{
+					PageSize: 10,
+				},
+			},
+			expectedList: []*common.RegistrationEntry{entry3, entry2, entry1},
+		},
+	}
+	for _, testCase := range tests {
+		testCase := testCase
+		s.T().Run(testCase.name, func(t *testing.T) {
+			resp, err := s.handler.ListAllEntriesWithPages(context.Background(), testCase.request)
+			if testCase.err != "" {
+				require.EqualError(t, err, testCase.err)
+				return
+			}
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+			expectedResponse := &datastore.ListRegistrationEntriesResponse{
+				Entries: testCase.expectedList,
+			}
+			util.SortRegistrationEntries(expectedResponse.Entries)
+			util.SortRegistrationEntries(resp.Entries)
+			require.Equal(t, expectedResponse.Entries, resp.Entries)
+		})
+	}
+
+	// test pagination tokens
+	resp, err := s.handler.ListAllEntriesWithPages(context.Background(), &registration.ListAllEntriesRequest{
+		Pagination: &registration.Pagination{
+			PageSize: 1,
+		},
+	})
+	s.Require().NoError(err)
+	s.Require().Len(resp.Entries, 1)
+	s.Require().True(proto.Equal(entry1, resp.Entries[0]))
+
+	// 2nd page
+	resp, err = s.handler.ListAllEntriesWithPages(context.Background(), &registration.ListAllEntriesRequest{
+		Pagination: &registration.Pagination{
+			Token:    resp.Pagination.Token,
+			PageSize: 1,
+		},
+	})
+	s.Require().NoError(err)
+	s.Require().Len(resp.Entries, 1)
+	s.Require().True(proto.Equal(entry2, resp.Entries[0]))
+
+	// 3rd page
+	resp, err = s.handler.ListAllEntriesWithPages(context.Background(), &registration.ListAllEntriesRequest{
+		Pagination: &registration.Pagination{
+			Token:    resp.Pagination.Token,
+			PageSize: 1,
+		},
+	})
+	s.Require().NoError(err)
+	s.Require().Len(resp.Entries, 1)
+	s.Require().True(proto.Equal(entry3, resp.Entries[0]))
+
+	// 4th page should be empty
+	resp, err = s.handler.ListAllEntriesWithPages(context.Background(), &registration.ListAllEntriesRequest{
+		Pagination: &registration.Pagination{
+			Token: resp.Pagination.Token,
+		},
+	})
+	s.Require().NoError(err)
+	s.Require().Len(resp.Entries, 0)
```

Comment:
Should we be asserting the token is empty? (suggesting no more pages)

Reply:
sure